http/query.go

```diff
@@ -259,17 +220,24 @@ func (r QueryRequest) proxyRequest(now func() time.Time) (*query.ProxyRequest, e
 	// Query is preferred over spec
 	var compiler flux.Compiler
 	if r.Query != "" {
-		compiler = lang.FluxCompiler{
-			Query: r.Query,
-		}
-	} else if r.AST != nil {
-		var err error
-		r.Spec, err = toSpec(r.AST, now)
+		pkg, err := flux.Parse(r.Query)
 		if err != nil {
 			return nil, err
 		}
-		compiler = lang.SpecCompiler{
-			Spec: r.Spec,
+		if r.Extern != nil {
```

Comment:
Same as my ASTCompiler note... I'd rather see extern as part of the SpecCompiler to decrease datastructure knowledge in the HTTP API.